mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoIncrementMethodGenerator.java

```diff
@@ -0,0 +1,190 @@
+  }
+
+  @Override
+  public boolean requiresReactive() {
+    // Validate the return type:
+    DaoReturnType returnType =
+        parseAndValidateReturnType(getSupportedReturnTypes(), Increment.class.getSimpleName());
+    if (returnType == null) {
+      return false;
+    }
+    return returnType.requiresReactive();
+  }
+
+  @Override
+  public Optional<MethodSpec> generate() {
+    // Validate the parameters:
+    // - the first one must be the entity.
+    // - a Function<BoundStatementBuilder, BoundStatementBuilder> can be added in second position.
+    List<? extends VariableElement> parameters = methodElement.getParameters();
+    VariableElement boundStatementFunction = findBoundStatementFunction(methodElement);
+    if (boundStatementFunction != null) {
+      parameters = parameters.subList(0, parameters.size() - 1);
+    }
+    TypeElement entityElement =
+        parameters.isEmpty()
+            ? null
+            : EntityUtils.asEntityElement(parameters.get(0), typeParameters);
+    if (entityElement == null) {
+      context
+          .getMessager()
+          .error(
+              methodElement,
+              processedType,
+              "%s methods must take the entity to update as the first parameter",
+              Increment.class.getSimpleName());
+      return Optional.empty();
+    }
+    warnIfCqlNamePresent(parameters.subList(0, 1));
+    EntityDefinition entityDefinition = context.getEntityFactory().getDefinition(entityElement);
+
+    // Validate the return type:
+    DaoReturnType returnType =
+        parseAndValidateReturnType(getSupportedReturnTypes(), Increment.class.getSimpleName());
+    if (returnType == null) {
+      return Optional.empty();
+    }
+
+    // Generate the method:
+    String helperFieldName = enclosingClass.addEntityHelperField(ClassName.get(entityElement));
+    String statementName =
+        enclosingClass.addPreparedStatement(
+            methodElement,
+            (methodBuilder, requestName) ->
+                generatePrepareRequest(
+                    methodBuilder, requestName, entityDefinition, helperFieldName));
+
+    CodeBlock.Builder updateStatementBlock = CodeBlock.builder();
+
+    updateStatementBlock.addStatement(
+        "$T boundStatementBuilder = $L.boundStatementBuilder()",
+        BoundStatementBuilder.class,
+        statementName);
+
+    populateBuilderWithStatementAttributes(updateStatementBlock, methodElement);
+    populateBuilderWithFunction(updateStatementBlock, boundStatementFunction);
+
+    String entityParameterName = parameters.get(0).getSimpleName().toString();
+
+    // All entity properties are bound in the generated request: primary key columns in the WHERE
+    // clause, and other columns in the SET clause.
+    updateStatementBlock.addStatement(
+        "$1L.set($2L, boundStatementBuilder, $3T.$4L)",
+        helperFieldName,
+        entityParameterName,
+        NullSavingStrategy.class,
+        NullSavingStrategy.DO_NOT_SET);
```

Comment:
If there are multiple counters in the table, some of the entity fields might left to zero, I didn't filter them out here. Cassandra handles 0-increments well, but I'm not sure if that has performance implications, from a quick look at the source I don't see anywhere where it would skip them. I'll ask around.

Alternatively:
* if the server version supports unset values, you can declare your fields as `Long` and leave them to null
* or you can have one separate entity class for each combination of counters to increment

I'll document that in the manual.

Reply:
This is moot point with new behavior: you can declare multiple methods with however many different combinations you need, the mapper will prepare a different CQL request for each.
```java
@Increment(entityClass = ProductRating.class)
void increment(UUID productId, long counter1, long counter2);

@Increment(entityClass = ProductRating.class)
void increment(UUID productId, long counter2, long counter3);
```